spacesvm/src/api/client.rs

```diff
@@ -23,13 +30,19 @@ pub use http::Uri;
 pub struct Client<C> {
     id: u64,
     client: HyperClient<C>,
-    pub uri: Uri,
+    endpoint: Uri,
+    private_key: Option<Key>,
 }
 
 impl Client<HttpConnector> {
-    pub fn new(uri: Uri) -> Self {
+    pub fn new(endpoint: Uri) -> Self {
```

Comment:
lets make this a &str and then validate as Uri

Reply:
maybe down the road client has a builder but for not I think how it is works fine.